### PR TITLE
feat: [CDS-78994]: Update KVTagInput to show create tag popover

### PR DESCRIPTION
**Pull request by rameshx**
This updates the KVTagInput to render a popover containing the `Create ${tag}` menu item. This improves the UX by letting the users know a tag is created only after pressing enter or clicking the `Create ${tag}` item.

https://github.com/harness/uicore/assets/52284933/8be47918-a0c9-4d47-9b42-d46ae70d29d5

### CODE REVIEW

**Review by sahithikolichala — DISMISSED** [review]

File: yarn.lock
Comment (rameshx) [inline yarn.lock:1]: This is generated after running `npm_config_yes=true npx yarn-audit-fix`. Getting this error without these changes:

```
10% building 0/1 entries 0/0 dependencies 0/0 modulesnode:internal/crypto/hash:71
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:130:10)
    at BulkUpdateDecorator.hashFactory (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/util/createHash.js:145:18)
    at BulkUpdateDecorator.update (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/util/createHash.js:46:50)
    at OriginalSource.updateHash (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack-sources/lib/OriginalSource.js:104:8)
    at NormalModule._initBuildHash (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/NormalModule.js:868:17)
    at handleParseResult (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/NormalModule.js:934:10)
    at /Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/NormalModule.js:1026:4
    at processResult (/Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/NormalModule.js:743:11)
    at /Users/rameshreddy/Documents/uicore/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/NormalModule.js:807:5
    at /Users/rameshreddy/Documents/uicore/node_modules/loader-runner/lib/LoaderRunner.js:406:3
    at iterateNormalLoaders (/Users/rameshreddy/Documents/uicore/node_modules/loader-runner/lib/LoaderRunner.js:232:10)
    at /Users/rameshreddy/Documents/uicore/node_modules/loader-runner/lib/LoaderRunner.js:223:4
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
error Command failed with exit code 1.
```

File: packages/uicore/src/components/FormikForm/FormikForm.tsx [hunk 1]
```diff
@@ -160,10 +160,57 @@ function KVTagInput(props: KVTagInputProps & FormikContextProps<any>) {
     label,
     ...rest
   } = restProps
+  const { separator = /[,\n\r]/ } = tagsProps ?? {}
```
Comment (rameshx) [inline packages/uicore/src/components/FormikForm/FormikForm.tsx:163]: `/[,\n\r]/` is the default separator that is used currently

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sahithikolichala — APPROVED** [review]

File: yarn.lock
Comment (abhinavrastogi-harness) [inline yarn.lock:1]: why do we have a change in the lockfile? seems unrelated.
looks like some new dependencies also got added.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

File: yarn.lock
Comment (rameshx) [reply yarn.lock:1]: This error is thrown when using the latest version of node 16, 16.20.2. Keeping this change as it is needed for everyone using the latest v16. Ran all unit tests in uicore and harness-core-ui to ensure this change is not breaking anything.

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**